jelares/portfolio/src/main/webapp/javascript/script.js

```diff
@@ -0,0 +1,153 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Changes the text of an element to a randomly chosen string
+ */
+function addRandomText(textChoices, elementId) {
+  // Pick a random text.
+  const textChoice =
+      textChoices[Math.floor(Math.random() * textChoices.length)];
+
+  // Add it to the page.
+  const textContainer1 = document.getElementById(elementId + '1');
+  const textContainer2 = document.getElementById(elementId + '2');
+
+  if (textContainer1.style.opacity === '0' &&
+      textContainer2.style.opacity === '0') {
+    textContainer1.innerText = textChoice;
+    textContainer2.style.opacity = 0;
+    textContainer1.style.opacity = 1;
+
+  } else if (textContainer1.style.opacity === '1') {
+    textContainer2.innerText = textChoice;
+    textContainer1.style.opacity = 0;
+    textContainer2.style.opacity = 1;
+
+  } else {
+    textContainer1.innerText = textChoice;
+    textContainer2.style.opacity = 0;
+    textContainer1.style.opacity = 1;
+  }
+}
+
+// Random Greeting
+function addRandomGreeting() {
+  addRandomText(
+      [
+        'Hola', 'Bonjour', 'Hello', 'Hej', 'Hallo', 'Greetings', 'Hey',
+        'Welcome'
+      ],
+      'greeting-container');
+}
+
+// Random photo subjects
+function addRandomPhoto() {
+  addRandomText(
+      [
+        'mountains', 'nature', 'the sea', 'my pets', 'cities', 'people',
+        'the world', 'kindness', 'the crazy', 'the new', 'the unbelievable'
+      ],
+      'photo-subject-container');
+}
+
+// Random photo subjects
+function addRandomProject() {
+  addRandomText(
+      ['us', 'me', 'you', 'humanity', 'friends', 'family', 'fun', 'learning'],
+      'project-subject-container');
+}
+
+// Random blog subjects
+function addRandomBlog() {
+  addRandomText(
+      [
+        'my parents', 'my coworkers', 'the fans', 'workers', 'dreamers',
+        'my cat', 'myself'
+      ],
+      'blog-subject-container');
+}
+
+/**
+ * Parallax elements scrolling function
+ */
+function parallaxElements() {
+  const targets = document.querySelectorAll('.parallax-landing');
+
+  for (let index = 0; index < targets.length; index++) {
+    let targetOffsetMiddle =
+        targets[index].offsetTop + Math.floor(targets[index].clientHeight / 2);
+    let windowOffsetMiddle =
+        window.pageYOffset + Math.floor(window.innerHeight / 2);
+
+    if (windowOffsetMiddle >= targetOffsetMiddle) {
+      let pos = (windowOffsetMiddle - targetOffsetMiddle) *
+          targets[index].dataset.rate;
```

Comment:
I believe `targetOffsetMiddle`, `windowOffsetMiddle`, and `pos` can be declared as `const` here, since they are never re-assigned within their scope. That is, the first two are scoped to this for-block, and are not reassigned within it, and `pos` is scoped to this if-block, and not reassigned within it.

After a scope is exited, I expect all of the local variables to be destroyed. On the next iteration, a new scope is created, and the variables are newly created (even though the variables have the same name in each iteration).

Reply:
Got it! That makes sense I'll update it